backend/v3/domain/settings.go

```diff
@@ -0,0 +1,286 @@
+	BackgroundColorDark string               `json:"backgroundColorDark,omitempty"`
+	WarnColorDark       string               `json:"warnColorDark,omitempty"`
+	FontColorDark       string               `json:"fontColorDark,omitempty"`
+	HideLoginNameSuffix bool                 `json:"hideLoginNameSuffix,omitempty"`
+	ErrorMsgPopup       bool                 `json:"errorMsgPopup,omitempty"`
+	DisableWatermark    bool                 `json:"disableMsgPopup,omitempty"`
+	ThemeMode           LabelPolicyThemeMode `json:"themeMode,omitempty"`
+
+	LabelPolicyLightLogoURL *string `json:"labelPolicyLightLogoURL,omitempty"`
+	LabelPolicyDarkLogoURL  *string `json:"labelPolicyDarkLogoURL,omitempty"`
+
+	LabelPolicyLightIconURL *url.URL `json:"labelPolicyLightIconURL,omitempty"`
+	LabelPolicyDarkIconURL  *url.URL `json:"labelPolicyDarkIconURL,omitempty"`
+
+	LabelPolicyFontURL *string `json:"labelPolicyLightFontURL,omitempty"`
+}
+
+type LabelSetting struct {
+	*Setting
+	Settings LabelSettings
+}
+
+type PasswordComplexitySettings struct {
+	IsDefault    bool   `json:"isDefault,omitempty"`
+	MinLength    uint64 `json:"minLength,omitempty"`
+	HasLowercase bool   `json:"hasLowercase,omitempty"`
+	HasUppercase bool   `json:"hasUppercase,omitempty"`
+	HasNumber    bool   `json:"hasNumber,omitempty"`
+	HasSymbol    bool   `json:"hasSymbol,omitempty"`
+}
+
+type PasswordComplexitySetting struct {
+	*Setting
+	Settings PasswordComplexitySettings
+}
+
+type PasswordExpirySettings struct {
+	IsDefault      bool   `json:"isDefault,omitempty"`
+	ExpireWarnDays uint64 `json:"expireWarnDays,omitempty"`
+	MaxAgeDays     uint64 `json:"maxAgeDays,omitempty"`
+}
+
+type PasswordExpirySetting struct {
+	*Setting
+	Settings PasswordExpirySettings
+}
+
+type LockoutSettings struct {
+	IsDefault           bool   `json:"isDefault,omitempty"`
+	MaxPasswordAttempts uint64 `json:"maxPasswordAttempts,omitempty"`
+	MaxOTPAttempts      uint64 `json:"maxOtpAttempts,omitempty"`
+	ShowLockOutFailures bool   `json:"showLockOutFailures,omitempty"`
+}
+
+type LockoutSetting struct {
+	*Setting
+	Settings LockoutSettings
+}
+
+type DomainSettings struct {
+	IsDefault                              bool `json:"isDefault,omitempty"`
+	UserLoginMustBeDomain                  bool `json:"userLoginMustBeDomain,omitempty"`
+	ValidateOrgDomains                     bool `json:"validateOrgDomains,omitempty"`
+	SMTPSenderAddressMatchesInstanceDomain bool `json:"smtpSenderAddressMatchesInstanceDomain,omitempty"`
+}
+
+type DomainSetting struct {
+	*Setting
+	Settings DomainSettings
+}
+
+type SecuritySettings struct {
+	Enabled               bool     `json:"enabled,omitempty"`
+	EnableIframeEmbedding bool     `json:"enableIframe_embedding,omitempty"`
+	AllowedOrigins        []string `json:"allowedOrigins,omitempty"`
+	EnableImpersonation   bool     `json:"enableImpersonation,omitempty"`
+}
+
+type SecuritySetting struct {
+	*Setting
+	Settings SecuritySettings
+}
```

Comment:
As we anyway need to touch the SettingsRepo I would change the `...Setting` structs like this example:


```suggestion
type SecuritySetting struct {
	Setting
	Enabled               bool     `json:"enabled,omitempty"`
	EnableIframeEmbedding bool     `json:"enableIframe_embedding,omitempty"`
	AllowedOrigins        []string `json:"allowedOrigins,omitempty"`
	EnableImpersonation   bool     `json:"enableImpersonation,omitempty"`
}
```

Reply:
@adlerhurst because of the fact we read the settings table then we unmarhsal the settings column, it makes more sense/is more performant to keep Setting as an embeded pointer